apps/main/src/loan/components/PageLlamaMarkets/cells/LineGraphCell.tsx

```diff
@@ -34,65 +28,40 @@ const calculateDomain =
-  const isPool = marketType == LlamaMarketType.Pool
-  const showMintGraph = !isPool && type === 'borrow'
-  const contractAddress = isPool ? controllerAddress : address
-  const params = { blockchainId: blockchainId, contractAddress }
-  const { data: poolSnapshots, isLoading: poolIsLoading } = useLendingSnapshots(params, isPool)
-  const { data: mintSnapshots, isLoading: mintIsLoading } = useCrvUsdSnapshots(params, showMintGraph)
-  if (isPool) {
-    return { snapshots: poolSnapshots, isLoading: poolIsLoading, snapshotKey: `${type}_apy` as const }
-  }
-  return { snapshots: showMintGraph ? mintSnapshots : null, isLoading: mintIsLoading, snapshotKey: 'rate' as const }
 }
 
 /**
  * Line graph cell that displays the average historical APY for a vault and a given type (borrow or lend).
  */
-export const LineGraphCell = ({ market, type, showChart }: LineGraphCellProps) => {
-  const { snapshots, snapshotKey, isLoading } = useSnapshots(market, type)
-  const currentValue = market.rates[type]
-  const rate = useMemo(
-    () => (snapshots?.length ? meanBy(snapshots, (row) => row[snapshotKey]) : currentValue),
-    [snapshots, currentValue, snapshotKey],
-  )
+export const LineGraphCell = ({ market, type }: RateCellProps) => {
+  const { snapshots, snapshotKey, isLoading, rate } = useSnapshots(market, type)
   const { design } = useTheme()
   if (rate == null) {
     return '-'
   }
-
   return (
-    <Stack direction="row" alignItems="center" justifyContent="end" gap={3} data-testid={`line-graph-cell-${type}`}>
-      {rate.toPrecision(4)}%
-      {showChart && (
-        <Box data-testid={`line-graph-${type}`}>
-          {snapshots?.length ? (
-            <LineChart data={snapshots} {...graphSize} compact>
-              <YAxis hide type="number" domain={calculateDomain(snapshots[0][snapshotKey])} />
-              <Line
-                type="monotone"
-                dataKey={snapshotKey}
-                stroke={getColor(design, snapshots, snapshotKey)}
-                strokeWidth={1}
-                dot={<></>}
-              />
-            </LineChart>
-          ) : isLoading ? (
-            <Skeleton {...graphSize} />
-          ) : (
-            <Typography sx={{ ...graphSize, alignContent: 'center', textAlign: 'left' }} variant="bodyXsBold">
-              {t`No historical data`}
-            </Typography>
-          )}
-        </Box>
+    <Box data-testid={`line-graph-${type}`}>
+      {snapshots?.length ? (
+        <LineChart data={snapshots} {...graphSize} compact>
+          <YAxis hide type="number" domain={calculateDomain(snapshots[0][snapshotKey] as number)} />
+          <Line
+            type="monotone"
+            dataKey={snapshotKey}
+            stroke={getColor(design, snapshots, snapshotKey)}
+            strokeWidth={1}
+            dot={<></>}
+          />
+        </LineChart>
+      ) : isLoading ? (
+        <Skeleton {...graphSize} />
```

Comment:
There's no need for the hardcoded `graphSize` when you use the linechart as its child; the skeleton will infer the size. You can see an example in `Metric.tsx`:
```tsx
      {loading ? (
        <Skeleton variant="text">
          <MetricValue {...metricValueProps} />
        </Skeleton>
      ) : (
        <MetricValue {...metricValueProps} />
      )}
```
You'll have to extract the LineChart above into a component that you can invoke twice. Or you can create a Skeleton helper component that basically just wraps its `children` property inside a `<Skeleton>` if the provided property `isLoading` is set to true. I didn't do that for the metric component, but I could've.

Reply:
graphSize was hardcoded for the chart lib, not necessarily for the skeleton